File: backend/app/schemas/hr_confirmation.py
"""HR 确认事项相关 Pydantic v2 schema。"""

from __future__ import annotations

from datetime import date, datetime
from typing import Any, Optional

from pydantic import BaseModel, Field


class ConfirmationItemResponse(BaseModel):
    """确认事项响应。"""
    id: int
    employee_id: int
    employee_name: Optional[str] = None
    employment_id: Optional[int] = None
    source_type: str
    source_id: Optional[str] = None
    issue_code: str
    title: str
    description: Optional[str] = None
    before_data: Optional[Any] = None
    after_data: Optional[Any] = None
    item_status: str
    created_at: datetime
    handled_at: Optional[datetime] = None


class ConfirmationListResponse(BaseModel):
    """确认事项列表响应。"""
    items: list[ConfirmationItemResponse]
    total: int
    pending_count: int


class ConfirmActionRequest(BaseModel):
    """确认/拒绝/忽略操作的请求体。"""
    note: Optional[str] = Field(default=None, description="备注")
    action_data: Optional[dict[str, Any]] = Field(
        default=None,
        description="业务动作数据，用于补充缺失字段等场景。"
        "例如 {\"actual_regularization_date\": \"2024-07-01\"}",
    )

File: frontend/src/types/hr-confirmation.ts
// HR 待确认事项相关的 TypeScript 类型定义

export interface ConfirmationItem {
  id: number
  employee_id: number
  employee_name?: string | null
  employment_id?: number | null
  source_type: string
  source_id?: string | null
  issue_code: string
  title: string
  description?: string | null
  before_data?: unknown
  after_data?: unknown
  item_status: 'PENDING' | 'CONFIRMED' | 'REJECTED' | 'IGNORED'
  created_at: string
  handled_at?: string | null
}

export interface ConfirmationListResponse {
  items: ConfirmationItem[]
  total: number
  pending_count: number
}

export interface ConfirmActionRequest {
  note?: string
  action_data?: Record<string, unknown>
}
